mullvad.openvpn-mullvad: 2.5.3 -> 2.6.0

TarCV opened the pull request:
###### Description of changes
Update OpenVPN inside Mullvad package to the version (2.6.0) required by the current Mullvad. Fixes #232794 

Also I've copied configure flags from Makefile ([1](https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L17-L20) [2](https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L35)) - as I understand it, Mullvad uses them when compiling their builds.
~~There are also additional flags they add only on Linux, not sure how to add them correctly (see https://github.com/mullvad/mullvadvpn-app-binaries/blob/main/Makefile#L35 ).~~
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
    **Tested against NixOS 23.05 I'm running on my system (510d721ce097150ae3b80f84b04b13b039186571)** 
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

TarCV committed:
mullvad.openvpn-mullvad: 2.5.3 -> 2.6.0

Update OpenVPN to the version required by current Mullvad

SuperSandro2000 commented on pkgs/applications/networking/mullvad/openvpn.nix:
```diff
@@ -16,68 +15,82 @@ openvpn.overrideAttrs (oldAttrs:
+        sha256 = "sha256-Via62wKVfMWHTmO7xIXXO7b5k0KYHs1D0JVg3qnXkeM=";
       })
     ];
+    postPatch = oldAttrs.postPatch or [ ] ++ [ "rm ./configure" ];
```
```suggestion
    postPatch = oldAttrs.postPatch or "" ++ ''
      rm ./configure
    '';
```

TarCV committed:
mullvad.openvpn-mullvad: Fix postPatch

TarCV committed:
mullvad.openvpn-mullvad: Move hooks to nativeBuildInputs

TarCV committed:
mullvad.openvpn-mullvad: Update configureFlags to match the flags from mullvadvpn-app-binaries on Linux too

TarCV commented:
Fixed. Also updated configureFlags to match the flags from mullvadvpn-app-binaries on Linux too.
I think this is all that was required for this update.

I see Borg haven't added cole-h as a reviewer and he is the maintainer of this package. So should I ping him?

TarCV commented:
@cole-h could you please review when you have time?

cole-h reviewed (approved):
If this fixes it for you, then let's get it in. I don't have any way to test right now (I'm not at my computer, nor do I use the OpenVPN tunnels).

SuperSandro2000 commented on pkgs/applications/networking/mullvad/openvpn.nix:
```diff
@@ -1,83 +1,108 @@
 
 openvpn.overrideAttrs (oldAttrs:
   let
+    inherit (lib) optional;
```
We are using this in two places where we could just have prefixed optional with lib..

arashsm79 commented:
I can also confirm that this has solved the issue with OpenVPN.
Would it be possible to backport this to 23.05?
